proto/types.proto

```diff
@@ -144,7 +144,14 @@ message FirecrackerDriveMount {
   // within the VM (i.e. ["rw", "noatime"]). Defaults to none if not specified. 
   repeated string Options = 4;
 
-  // TODO support for ratelimiters and read-only drives (issue #296)
+  // (Optional) RateLimiter configuration that will be applied to the
+  // backing-drive for the VM's rootfs
+  FirecrackerRateLimiter RateLimiter = 5;
+
+  // (Optional) If set to true, IsReadOnly results in the backing file for the
+  // drive being opened as read-only by the Firecracker VMM on the host, preventing
+  // any writes to the image from within the guest.
+  bool IsReadOnly = 6;
```

Comment:
Can we negate the flag to IsWritable? Mostly because FirecrackerRootDrive has that.

Reply:
Good call, updated. As a result I needed to add a little more functionality though because Linux will default a mount to "rw" if not otherwise specified. So this meant that if a DriveMount was read-only by default then the user would get an error unless they explicitly set "ro" in their mount options, which kind of sucks. So I updated the implementation to check that the `IsWritable` field matches what was specified in the mount options or, if there was no ro/rw option specified, explicitly sets it on behalf of the user.